src/commands/optimize.ts

```diff
@@ -84,25 +84,32 @@ export default class Optimize extends Command {
     this.optimizations = flags.optimization as Optimizations[];
     this.outputMethod = flags.output as Outputs;
 
-    this.metricsMetadata.optimizations = this.optimizations;
+    this.metricsMetadata.optimized = true;
 
     if (!(report.moveToComponents?.length || report.removeComponents?.length || report.reuseComponents?.length)) {
       this.log(`No optimization has been applied since ${this.specFile.getFilePath() ?? this.specFile.getFileURL()} looks optimized!`);
+      this.metricsMetadata.optimized = false;
       return;
     }
-
+    
     const isTTY = process.stdout.isTTY;
     if (this.isInteractive && isTTY) {
       await this.interactiveRun(report);
     }
 
     try {
```

Comment:
I think we could include this try-catch block in a separate function to cover the non-interactive scenario, so that the cognitive complexity of `run()` function is decreased (right now is above the max allowed). WDYT? @smoya